device/protocol_mqtt_v1.go

```diff
@@ -297,6 +297,53 @@ func (d *Device) SendAggregateMessage(interfaceName, interfacePath string, value
 	return d.SendAggregateMessageWithTimestamp(interfaceName, interfacePath, values, time.Time{})
 }
 
+// SetProperty sets a  property in Astarte. It adds to the publishing channel a property set
+// message that is equivalent to `SendIndividualMessage` in every regard, except this call will fail
+// if the interface isn't a property interface. It is advised to use this call, as APIs might change in
+// the future. Once the property is set, it is also added to the local cache and can be retrieved at any time
+// using `GetProperty` or `GetAllProperties`.
+// This call can be blocking if the channel is full (see `MaxInflightMessages`).
+func (d *Device) SetProperty(interfaceName, path string, value interface{}) error {
+	// Get the interface from the introspection
+	iface, ok := d.interfaces[interfaceName]
+	if ok {
+		if iface.Type != interfaces.PropertiesType {
+			return fmt.Errorf("SetProperty can be used only on Property Interfaces", interfaceName)
+		}
+		// Validate the message
+		if err := interfaces.ValidateIndividualMessage(iface, path, value); err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("Interface %s not registered", interfaceName)
+	}
+	// We are good to go. Let's add the message to the channel.
+	return d.enqueueMqttV1Message(iface, path, value, time.Time{})
+}
+
+// DeleteProperty deletes an existing property from Astarte. It adds to the publishing channel a property deletion
+// message that contains the interface name and the path to be deleted. Once the property is deleted, it is also
+// removed from the local cache.
+// This call can be blocking if the channel is full (see `MaxInflightMessages`).
+func (d *Device) DeleteProperty(interfaceName, path string) error {
+	// Get the interface from the introspection
+	iface, ok := d.interfaces[interfaceName]
+	if ok {
+		if iface.Type != interfaces.PropertiesType {
+			return fmt.Errorf("SetProperty can be used only on Property Interfaces", interfaceName)
+		}
+		// Validate the path
+		if err := interfaces.ValidateInterfacePath(iface, path); err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("Interface %s not registered", interfaceName)
+	}
+	// We are good to go. Let's add the message to the channel.
```

Comment:
> Moreover, I think it would be cleaner to check interface ownership here, and not inside enqueueRawMqttV1Message.

The reason why I'm doing it in that function is that the ownership check applies to all sent messages, so I think it is where it belongs